### PR TITLE
PST-014-create-input-component

### PR DESCRIPTION
**Pull request by skrzypczyk-pawel**

### CODE REVIEW

File: src/components/StyledInput.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  label,
+  onBlur,
+  onFocus,
+  placeholder,
+  style,
+  type = "text",
+  value = "",
+}) => {
+  const [inputValue, setInputValue] = useState(value);
+  const [isFocused, setIsFocused] = useState(focused);
```
Comment (dominikszczuka) [inline src/components/StyledInput.tsx]: Use typing => useState<type>()

File: src/components/StyledInput.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  const handleFocus = () => {
+    setIsFocused(true);
+    if (onFocus) {
+      onFocus();
```
Comment (dominikszczuka) [inline src/components/StyledInput.tsx]: You can use Optional chaining operator => onFocus?.()

File: src/components/StyledInput.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+  };
+  const handleBlur = () => {
+    setIsFocused(false);
+    if (onBlur) {
```
Comment (dominikszczuka) [inline src/components/StyledInput.tsx]: You can use Optional chaining operator => onBlur?.()

File: src/components/StyledInput.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      onBlur();
+    }
+  };
+
+  return (
+    <div>
```
Comment (dominikszczuka) [inline src/components/StyledInput.tsx]: If you don't add style for this div I think you can use React.Fragment

File: src/components/StyledInput.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+import { colors } from "styles/palette";
+
+interface Props {
+  value: string;
+  error?: string;
+  focused?: boolean;
+  onBlur?: () => void;
+  onFocus?: () => void;
+  label?: string;
+  placeholder?: string;
```
Comment (dominikszczuka) [inline src/components/StyledInput.tsx:14]: I think you always set placeholder then is it good idea that it optional ?

Comment (Stared55) [reply src/components/StyledInput.tsx:14]: It can stay as it is

File: src/components/StyledInput.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+const StyledInput: FC<Props> = ({
+  error,
+  focused = false,
+  label,
+  onBlur,
+  onFocus,
+  placeholder,
+  style,
+  type = "text",
+  value = "",
```
Comment (Stared55) [inline src/components/StyledInput.tsx]: No. Value is not an optional property so you shouldn't set default value for this.

File: src/components/StyledInput.tsx [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  input: {
+    transition: "0.5s",
+    margin: 2,
+    boxShadow: `3px 3px 5px ${colors.grayBlue}`,
+    border: `1px solid ${colors.white}`,
+    outline: "none",
+    borderRadius: 5,
+    backgroundColor: colors.white,
+  },
+  inputFcs: {
```
Comment (Stared55) [inline src/components/StyledInput.tsx]: `focused` is enough

File: src/components/StyledInput.tsx [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    ":focus": {
+      boxShadow: `3px 3px 5px ${colors.skyBlue}`,
+      backgroundColor: colors.vLightBlue,
```
Comment (Stared55) [inline src/components/StyledInput.tsx]: What this 'v' means ?

File: src/components/StyledInput.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  };
+
+  return (
+    <div>
+      <label htmlFor={label}>
+        {label}
+        <input
+          id={label}
+          onFocus={handleFocus}
+          onChange={(event) => setInputValue(event.target.value)}
```
Comment (Stared55) [inline src/components/StyledInput.tsx]: onChange={onChange}
 
 and this handler should be passed from the parent component. Also move `const [inputValue, setInputValue] = useState(value);` into parent

File: src/screens/HomeScreen.tsx [hunk 10]
```diff
@@ -10,24 +10,38 @@ import { i18n } from "locale";
...
+  const [inputValue2, setInputValue2] = useState<string>("");
   return (
     <ScreenWrapper>
       <div className={css(styles.homeScreen)}>
         <h3>{i18n.t("header:title")}</h3>
-        <StyledInput value="" placeholder="Search..." />
-        <StyledInput value="" placeholder="Error..." error="Error Text!!!" />
+        <StyledInput
+          value={inputValue1}
+          onChange={(event) => setInputValue1(event.target.value)}
```
Comment (Stared55) [inline src/screens/HomeScreen.tsx]: ok, but u should pass a callback to onChange, so create a function above return , for example: 
`const handleInputValue = (value) => {  setInputValue1(value) }`
and pass it like that: onChange={handleInputValue}

**Review by Stared55 — APPROVED** [review]
✅

**Pull request merged**